main/inc/ajax/admin.ajax.php

```diff
@@ -35,6 +35,74 @@
     case 'version':
         echo version_check();
         break;
+
+    case 'save_block_extra':
+        $content = isset($_POST['content']) ? Security::remove_XSS($_POST['content']) : null;
+        $blockId = isset($_POST['block']) ? Security::remove_XSS($_POST['block']) : null;
```

Comment:
Watch out! $blockId is a string, so it would be better to call it "$blockName" or something like that.
Because you use $blockId to form part of a path to a file, it could be used as a path traversal vulnerability (for example to get to /etc/passwd). Make sure you use Security::check_rel_path() to compare the final path to api_get_path(SYS_PATH).'home/'